### PR TITLE
Ma/response mode

### PR DESCRIPTION
**Pull request by markusahlstrand**
We didn't have a common way of handling the response mode before, but handled each of the cases in different places. I'm thinking that the flow is pretty much:
1. authenticate
2. generate tokens
3. create a response.

I think this pr doesn't handle all the cases yet. For instance I'm guessing that the iframe probably should be the WEB_MESSAGE. Not really sure what the FORM_POST is used for...

### CODE REVIEW

File: src/authentication-flows/social.ts [hunk 1]
```diff
@@ -109,30 +110,18 @@ export async function socialAuthCallback({
...
+  const tokenResponse = await generateAuthResponse({
+    env,
+    userId,
+    sid: sessionId,
+    state: state.authParams.state,
+    nonce: state.authParams.nonce,
+    authParams: state.authParams,
+    user: profile,
+    responseType:
+      state.authParams.response_type || AuthorizationResponseType.TOKEN,
```
Comment (markusahlstrand) [inline src/authentication-flows/social.ts:123]: Defaulting here.. Maybe it's better to handle in the genereateAuthResponse function? What is the correct default by the way?

File: test/authentication-flows/ticket.spec.ts [hunk 2]
```diff
@@ -60,10 +63,7 @@ describe("passwordlessAuth", () => {
     expect(hashParams.get("id_token")).toBe(null);
     expect(hashParams.get("token_type")).toBe("Bearer");
     expect(hashParams.get("expires_in")).toBe("28800");
-    expect(hashParams.get("state")).toBe(encodeURIComponent(state));
-    expect(hashParams.get("state")).toBe(
-      "redirect_uri%3Dhttps%253A%252F%252Fexample.com%26client_id%3Dkvartal%26response_type%3Dcode%26state%3D12345678",
-    );
+    expect(hashParams.get("state")).toBe(state);
```
Comment (danielgent) [inline test/authentication-flows/ticket.spec.ts:66]: why revert this test @markusahlstrand ? :thinking:

File: src/authentication-flows/social.ts (hunk 1 above)
Comment (danielgent) [reply src/authentication-flows/social.ts:119]: Is there a correct default or are we just always setting `token` as the default :shrug:

**Review by danielgent — APPROVED** [review]

File: test/authentication-flows/ticket.spec.ts (hunk 2 above)
Comment (markusahlstrand) [reply test/authentication-flows/ticket.spec.ts:66]: Ahh... so the test was really that it should encode the state param. Maybe we should add a new test for that describing that it's what we are testing? I guess it doesn't today for some reason?

Comment (danielgent) [reply test/authentication-flows/ticket.spec.ts:66]: I have no idea Markus! Now everything is working :smile: 

I'm happy to forget that this ever happened :see_no_evil:

**Pull request merged**